olive/resource_path.py

```diff
@@ -0,0 +1,379 @@
+        config of the specified type.
+
+        :param resource_path: A string, a Path, or a dict.
+        :return: A resource path.
+        """
+        if isinstance(resource_path, ConfigBase):
+            return validate_config(resource_path, cls)
+
+        if isinstance(resource_path, dict):
+            return cls(**resource_path)
+
+        if isinstance(resource_path, Path) and not resource_path.exists():
+            ValueError(f"Resource path {resource_path} of type Path is not a file or folder.")
+
+        # check if the resource path is a file, folder, or a string name
+        type: ResourceType = None
+        config_key = "path"
+        if Path(resource_path).is_file():
+            type = ResourceType.LocalFile
+        elif Path(resource_path).is_dir():
+            type = ResourceType.LocalFolder
+        else:
+            type = ResourceType.StringName
+            config_key = "name"
+
+        logger.debug(f"Resource path {resource_path} is inferred to be of type {type}.")
+        return cls(type=type, config={config_key: resource_path})
+
+    def get_path(self) -> str:
+        """Return the resource path as a string."""
+        return self.config.get_path()
+
+    def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: bool = False) -> str:
+        """Save the resource to a directory."""
+        return self.config.save_to_dir(dir_path, name, overwrite)
+
+    def is_local_resource(self) -> bool:
+        """Return True if the resource is a local resource."""
+        return self.type in LOCAL_RESOURCE_TYPES
+
+
+def _overwrite_helper(new_path: Union[Path, str], overwrite: bool):
+    new_path = Path(new_path).resolve()
+
+    # check if the resource already exists
+    if new_path.exists():
+        if not overwrite:
+            # raise an error if the resource already exists
+            raise FileExistsError(f"File {new_path} already exists and overwrite is set to False.")
+        else:
+            # delete the resource if it already exists
+            if new_path.is_file():
+                new_path.unlink()
+            else:
+                shutil.rmtree(new_path)
+
+
+class LocalResourceConfig(ResourceConfig):
+    path: Union[Path, str] = Field(..., description="Path to the resource.")
+
+    def get_path(self) -> str:
+        return str(self.path)
+
+    def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: bool = False) -> str:
+        # directory to save the resource to
+        dir_path = Path(dir_path).resolve()
+        dir_path.mkdir(parents=True, exist_ok=True)
+
+        # path to save the resource to
+        if name:
+            new_path_name = Path(name).with_suffix(self.path.suffix).name
+        else:
+            new_path_name = self.path.name
+        new_path = dir_path / new_path_name
+        _overwrite_helper(new_path, overwrite)
```

Comment:
If overwrite is false and the file is already dumped to local file system, should be raise the exception to stop olive?
How about to catch the exception from this method?

Reply:
I think it is better to fail here? We cannot guarantee that the file present already in the save_dir is the same as the file we want to download. 
The olive engine will never encounter this failure since it handles model download using cache and also sets `overwrite` to `True` when this method is used.

Reply:
I originally created the `_overwrite_helper` function to abstract common code that is used by all resource path types. so this logic would otherwise been directly inside the `save_to_dir` function. 

I can rename the `_overwrite_helper` function if that helps? With regards to the behavior, I think we should fail if overwrite=False and a similarly named file/folder already exists.

Reply:
I do not touch too much detailed code. But what if if path exists, we force the overwrite to true?

Reply:
if `overwrite` is true and file/folder exists, we delete the existing resource and write the new one to the same path. `_overwrite_helper` does the deletion. the new files are saved by the code after the `_overwrite_helper` is called.